src/course-outline/CourseOutline.jsx

```diff
@@ -85,20 +85,36 @@ const CourseOutline = ({ courseId }) => {
     handleDuplicateSubsectionSubmit,
     handleNewSectionSubmit,
     handleNewSubsectionSubmit,
-    handleDragNDrop,
+    handleSectionDragAndDrop,
+    handleSubsectionDragAndDrop,
   } = useCourseOutline({ courseId });
 
   const [sections, setSections] = useState(sectionsList);
 
-  const initialSections = [...sectionsList];
+  let initialSections = [...sectionsList];
 
   const {
     isShow: isShowProcessingNotification,
     title: processingNotificationTitle,
   } = useSelector(getProcessingNotification);
 
   const finalizeSectionOrder = () => (newSections) => {
-    handleDragNDrop(newSections.map((section) => section.id), () => {
+    initialSections = [...sectionsList];
```

Comment:
Why is this being updated here? 

I see that we're copying the sectionsList and keeping initialSections up to date with the latest value of sectionsList. So why not just use sectionsList? 

i.e. removeinitialSections and just use sectionsList wherever you're currently using initialSections.